app/src/main/report/components/ReportsList/ReportsList.ts

```diff
@@ -0,0 +1,50 @@
+import { connect } from 'react-redux';
+import { Dispatch } from "redux";
+import { compose, lifecycle, branch, renderComponent } from "recompose";
+import { clone } from "lodash";
+
+import {ReportAppState} from "../../reducers/reportAppReducers";
+import { reportsActions } from "../../actions/reportsActions";
+import {ReportsListComponent, ReportsListComponentProps} from "./ReportsListComponent";
+import {Report} from "../../../shared/models/Generated";
+import {ReportsSortingFields} from "../../actionTypes/ReportsActionsTypes";
+import { LoadingElement } from "../../../shared/partials/LoadingElement/LoadingElement";
+
+export interface ReportsListContainerProps extends ReportsListComponentProps {
+    getReports: () => void;
+    ready: boolean;
+}
+
+export const getDisplayedReportsList = (items: Report[], sortBy: ReportsSortingFields) => {
+    if (!items || !items.length) return [];
+    let displayItemsSorted = clone(items).sort(sortReports(sortBy));
+    return displayItemsSorted;
+}
+
+export const sortReports = (sortBy: ReportsSortingFields) => (a: Report, b: Report)=> (a[sortBy] < b[sortBy] ? -1 : 1);
+
+const lifecyleProps = {
+    componentDidMount() {
+        this.props.getReports();
+    }
+}
+
+export const mapStateToProps = (state: ReportAppState): Partial<ReportsListContainerProps> => {
+    return {
+        reports: getDisplayedReportsList(state.reports.reports, state.reports.reportsSortBy),
+        ready: Array.isArray(state.reports.reports)
+    }
+};
+
+export const mapDispatchToProps = (dispatch: Dispatch<ReportAppState>): Partial<ReportsListContainerProps> => {
+    return {
+        getReports: () => dispatch(reportsActions.getReports())
+    }
+};
+
+const enhance = compose(
+    connect(mapStateToProps, mapDispatchToProps),
+    lifecycle(lifecyleProps),
```

Comment:
My only qualm here is that now we have 2 styles of component - this style and the component class style. We need to choose whether or not we're going to use `recompose` and make sure we create components consistently going forwards.

I feel like I need to write a component, write some tests etc myself to really get a feel for whether or not it's a better pattern. I propose we give ourselves 1 sprint to try it out, and at the end we decide to either use `recompose` everywhere or not at all. Looping @MartinEden in on this discussion too, when he comes back online!